### ISSUE TITLE
Creating a new DataLoader per request

### ISSUE THREAD

**Issue by sibelius** [entria/graphql-dataloader-boilerplate#11]
Based on discussion on slack with @ThisIsMissEm and @xuorig it is better to have one dataloader per request

**Comment by plraphael** [comment]
Can you summarize the discussion here @sibelius?

**Comment by sibelius** [comment]
Right now we keep use the same DataLoaders for all requests: https://github.com/sibelius/graphql-dataloader-boilerplate/blob/master/src/loader/UserLoader.js#L21 as it is a static

It is a common pattern to create a new DataLoader per request (https://github.com/facebook/dataloader#creating-a-new-dataloader-per-request)

> In many applications, a web server using DataLoader serves requests to many different users with different access permissions. It may be dangerous to use one cache across many users, and is encouraged to create a new cache per request:

Another problem is that if we have 2 or more server running this GraphQL server, each server will have their own version of DataLoader, so when one of them clear the cache of an item, it will only clear the cache of their own DataLoader, and not updating the cache of other DataLoaders
